### PR TITLE
[Coral-Trino] Fix substr start index issue

### PR DESCRIPTION
**Pull request by jerryleooo**
### What changes are proposed in this pull request, and why are they necessary?

Fix https://github.com/linkedin/coral/issues/433

### How was this patch tested?

Modify the existing unit tests accordingly.

### CODE REVIEW

**Comment by jerryleooo** [comment]
Hi @wmoustafa can help have a review? Thanks

**Comment by jerryleooo** [comment]
@wmoustafa @findepi can have a look?

**Review by ljfgem — COMMENTED** [review]
Thank you @jerryleooo for this PR!

**Comment by ljfgem** [comment]
And I think `coral-dbt/src/main/resources/tests/__pycache__/test_incremental.cpython-311.pyc` is not needed?

File: coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CoralToTrinoSqlCallConverter.java [hunk 1]
```diff
@@ -72,7 +72,16 @@ protected SqlCall transform(SqlCall sqlCall) {
...
         // string functions
-        new OperatorRenameSqlCallTransformer(SqlStdOperatorTable.SUBSTRING, 2, "SUBSTR"),
+        new JsonTransformSqlCallTransformer(SqlStdOperatorTable.SUBSTRING, 2, "substr",
+            "[{\"input\": 1}, {\"op\": \"+\", \"operands\": [{\"input\": 2}, {\"value\": 1}]}]", null, null),
+        new JsonTransformSqlCallTransformer(SqlStdOperatorTable.SUBSTRING, 2, "substr",
+            "[{\"input\": 1}, {\"op\": \"+\", \"operands\": [{\"input\": 2}, {\"value\": 1}]}]", null, null),
+        new JsonTransformSqlCallTransformer(SqlStdOperatorTable.SUBSTRING, 3, "substr",
+            "[{\"input\": 1}, {\"op\": \"+\", \"operands\": [{\"input\": 2}, {\"value\": 1}]}, {\"input\": 3}]", null,
+            null),
+        new JsonTransformSqlCallTransformer(hiveToCoralSqlOperator("substr"), 3, "substr",
```
[the remark this answers is not in the document]
Comment (jerryleooo) [reply coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CoralToTrinoSqlCallConverter.java:82]: fixed

File: coral-trino/src/test/java/com/linkedin/coral/trino/rel2trino/HiveToTrinoConverterTest.java [hunk 2]
```diff
@@ -162,7 +162,7 @@ public Object[][] viewTestCasesProvider() {
...
             + "FROM \"test\".\"table_from_utc_timestamp\" AS \"table_from_utc_timestamp\"" },
 
-        { "test", "date_calculation_view", "SELECT \"date\"(CAST(\"substr\"('2021-08-20', 1, 10) AS TIMESTAMP)), \"date\"(CAST('2021-08-20' AS TIMESTAMP)), \"date\"(CAST('2021-08-20 00:00:00' AS TIMESTAMP)), \"date_add\"('day', 1, \"date\"(CAST('2021-08-20' AS TIMESTAMP))), \"date_add\"('day', 1, \"date\"(CAST('2021-08-20 00:00:00' AS TIMESTAMP))), \"date_add\"('day', 1 * -1, \"date\"(CAST('2021-08-20' AS TIMESTAMP))), \"date_add\"('day', 1 * -1, \"date\"(CAST('2021-08-20 00:00:00' AS TIMESTAMP))), CAST(\"date_diff\"('day', \"date\"(CAST('2021-08-21' AS TIMESTAMP)), \"date\"(CAST('2021-08-20' AS TIMESTAMP))) AS INTEGER), CAST(\"date_diff\"('day', \"date\"(CAST('2021-08-19' AS TIMESTAMP)), \"date\"(CAST('2021-08-20' AS TIMESTAMP))) AS INTEGER), CAST(\"date_diff\"('day', \"date\"(CAST('2021-08-19 23:59:59' AS TIMESTAMP)), \"date\"(CAST('2021-08-20 00:00:00' AS TIMESTAMP))) AS INTEGER)\n"
```
[the remark this answers is not in the document]
Comment (jerryleooo) [reply coral-trino/src/test/java/com/linkedin/coral/trino/rel2trino/HiveToTrinoConverterTest.java:165]: fixed

**Comment by jerryleooo** [comment]
add `.pyc` into `.gitignore`, so no `.pyc` file will be included in commit

**Comment by ebyhr** [comment]
@ljfgem Gentle reminder.

**Comment by wmoustafa** [comment]
@ljfgem is this pending anything or can we merge?

**Comment by ljfgem** [comment]
@jerryleooo @ebyhr  new version with the fix 2.2.23 has been released. Thanks for your contributions.

**Pull request merged**